cdap-ui/cypress/integration/pipeline.spec.ts

```diff
@@ -96,13 +112,88 @@ describe('Creating a pipeline', () => {
       .find('.toggle-switch')
       .click();
     cy.get('@instrumentationDiv').contains('Off');
+
+    cy.contains('Pipeline alert').click();
+    cy.contains('+').click();
+    cy.contains('Send Email').click();
+
+    // enter sender, recipients, subject, message
+
+    cy.get('[data-cy="sender"]').within(() => {
+      cy.get('input')
+        .scrollIntoView()
+        .focus()
+        .type(TEST_SENDER);
+    });
+    cy.get('[data-cy="recipients"]').within(() => {
+      cy.get('input').type(TEST_RECIPIENT);
+    });
+    cy.get('[data-cy="subject"]').within(() => {
+      cy.get('input').type(TEST_SUBJECT);
+    });
+    // validate and see error
+    cy.get('[data-cy="validate-btn"]').click();
+
+    cy.contains('error found').should('exist');
+
+    // Fix missing field to resolve error
+    cy.get('[data-cy="message"]').within(() => {
+      cy.get('button').click();
+    });
+    // validate
+    cy.get('[data-cy="validate-btn"]').click();
+    cy.wait(2000);
+    cy.contains('error found').should('not.exist');
+
+    // click next
+    cy.get('[data-cy="next-btn"]').click();
+
+    // click confirm
+    cy.get('[data-cy="confirm-btn"]').click();
+
+    // See email alert
+    cy.get('[data-cy="saved-alerts"]').within(() => {
+      cy.contains('Email').should('exist');
+    });
+
     cy.get('[data-testid=config-apply-close]').click();
+  });
+  it('should run or not run Preview based on user action on postrun action modal', () => {
```

Comment:
nit: it would be clear if the test case says either one or the other. Saying `should run or not run preview` doesn't say what this test case is testing :). May be split it into two test cases one starting preview after clicking yes and one not starting preview because user clicked no.

Reply:
That makes a lot of sense! Have split into two tests :)